### ISSUE TITLE
:warning: Action Required: Replace Deprecated gcr.io/kubebuilder/kube-rbac-proxy

### ISSUE THREAD

**Issue by camilamacedo86** [ansible/eda-server-operator#262]
## Description

:warning: **The image `gcr.io/kubebuilder/kube-rbac-proxy` is deprecated and will become unavailable.**
**You must move as soon as possible, sometime from early 2025, the GCR will go away.**

> _Unfortunately, we're unable to provide any guarantees regarding timelines or potential extensions at this time. Images provided under GRC will be unavailable from March 18, 2025, [as per announcement](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr). However, `gcr.io/kubebuilder/`may be unavailable before this date due [to efforts to deprecate infrastructure](https://github.com/kubernetes/k8s.io/issues/2647)._

- **If your project uses `gcr.io/kubebuilder/kube-rbac-proxy`, it will be affected.**
  Your project may fail to work if the image cannot be pulled. **You must take action as soon as possible.**

- However, if your project is **no longer using this image**, **no action is required**, and you can close this issue.

## Using the image `gcr.io/kubebuilder/kube-rbac-proxy`?

[kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) was historically used to protect the metrics endpoint. However, its usage has been discontinued in [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder). The default scaffold now leverages the [`WithAuthenticationAndAuthorization`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/metrics/filters#WithAuthenticationAndAuthorization) feature provided by [Controller-Runtime](https://github.com/kubernetes-sigs/controller-runtime).

This feature provides integrated support for securing metrics endpoints by embedding authentication (`authn`) and authorization (`authz`) mechanisms directly into the controller manager's metrics server, replacing the need for (https://github.com/brancz/kube-rbac-proxy) to secure metrics endpoints.

### What To Do?

**You must replace the deprecated image `gcr.io/kubebuilder/kube-rbac-proxy` with an alternative approach. For example:**
- Update your project to use [`WithAuthenticationAndAuthorization`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/metrics/filters#WithAuthenticationAndAuthorization):
  > _You can fully upgrade your project to use the latest scaffolds provided by the tool or manually make the necessary changes. Refer to the [FAQ and Discussion](https://github.com/kubernetes-sigs/kubebuilder/discussions/3907) for detailed instructions on how to manually update your project and test the changes._
- Alternatively, replace the image with another trusted source at your own risk, as its usage has been discontinued in Kubebuilder.

**For further information, suggestions, and guidance:**
- 📖 [FAQ and Discussion](https://github.com/kubernetes-sigs/kubebuilder/discussions/3907)
- 💬 Join the Slack channel: [#kubebuilder](https://communityinviter.com/apps/kubernetes/community#kubebuilder).

> **NOTE:** _This issue was opened automatically as part of our efforts to identify projects that might be affected and to raise awareness about this change within the community. If your project is no longer using this image, **feel free to close this issue**._

We sincerely apologize for any inconvenience this may cause.

**Thank you for your cooperation and understanding!** :pray:

**Comment by Francisco-xiq** [comment]
Hi there, I encountered this problem in my environment and just started working on a solution, kindly assign it to me if it's available. Please let me know if there are any specific guidelines I should follow.

**Comment by Francisco-xiq** [comment]
I believe there is a problem left behind in the repository.

The current release **1.0.2** points to a deprecated/non-existent `kube-rbac-proxy` image in `gcr.io` inside the generated `operator.yaml`. Not only that, but the `/config/testing/kustomization.yaml` points to a non-existent file: `../default/manager_auth_proxy_patch.yaml.`

So, anyone using the release `1.0.2` is going to have a problem deploying the system, and also anyone who tries to use the `config/default` in their `kustomization.yaml` like this:

```
resources:
  - https://github.com/ansible/eda-server-operator/config/default?ref=1.0.2

```
To solve this problem, we can remove the line in `config/testing/kustomization.yaml` so it stops pointing to a deleted resource. Looking through the commit logs, I see that the RBAC proxy was already replaced with a native solution (in commit `b115c38403`) in the main branch, so the deployment problem for end users could be solved just by releasing a new version of the eda-server-operator.

What you think?

**Comment by ttuffin** [comment]
Hi @Francisco-xiq,

There was [an announcement](https://forum.ansible.com/t/upcoming-changes-to-galaxy-and-eda-operator-releases/42285) over on the Ansible Forums regarding a change to operator releases. In short, we have stopped making tagged releases on Github, however you can still deploy a specific version of the operator by referencing the appropriate git commit. You can find more information about the change in the above link.

**Comment by Francisco-xiq** [comment]
Thanks for the explanation, ttuffin. I'm new to this community, so I'm still learning. As I mentioned, there is still a reference to something deprecated on the main branch in `config/testing/kustomization.yml`. I think we just need to remove `../default/manager_auth_proxy_patch.yaml`. Should I tackle that? If not, I guess this issue is resolved.

I've successfully deployed a test environment to work on new issues.

**Comment by ttuffin** [comment]
@Francisco-xiq you are correct, that file was removed in #323. If you would like to open a PR to remove that reference that would be great, thanks!

**Comment by ptoscano** [comment]
The initial issue (`kube-rbac-proxy`) was fixed with #323 -- linking it to this github issue.
